src/components/colorscale/calc.js

```diff
@@ -20,7 +20,7 @@ module.exports = function calc(trace, vals, containerStr, cLetter) {
 
     if(containerStr) {
         container = Lib.nestedProperty(trace, containerStr).get();
-        inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
+        inputContainer = Lib.nestedProperty(trace, containerStr).get();
```

Comment:
> The technical difficulty is that this object, on which it attempts to set cmin/cmax may not exist:

In what scenario. `calcColorscale` is only called if `hasMarkers(trace)` is true (and if `hasLines(trace)` ?